src/java/src/main/native/ai_onnxruntime_genai_Tensor.cpp

```diff
@@ -18,7 +18,7 @@ JNIEXPORT jlong JNICALL
 Java_ai_onnxruntime_genai_Tensor_createTensor(JNIEnv* env, jobject thiz, jobject tensor_data,
                                               jlongArray shape_dims_in, jint element_type_in) {
   void* data = env->GetDirectBufferAddress(tensor_data);
-  const int64_t* shape_dims = env->GetLongArrayElements(shape_dims_in, /*isCopy*/ 0);
+  const int64_t* shape_dims = reinterpret_cast<const int64_t*>(env->GetLongArrayElements(shape_dims_in, /*isCopy*/ 0));
```

Comment:
Wouldn't a reinterpret_cast remove type safety? Since now GetLongArrayElements could return a mismatched type and this won't be caught.

Reply:
This never works on macOS. On other platform it works.